lib/serializers.php

```diff
@@ -25,3 +60,165 @@ function valueObject(Writer $writer, $valueObject, $namespace) {
+ *
+ * @param Writer $writer
+ * @param array $items A list of items sabre/xml can serialize.
+ * @param string $childElementName Element name in clark-notation
+ * @return array
+ */
+function repeatingElements(Writer $writer, array $items, $childElementName) {
+
+    foreach ($items as $item) {
+        $writer->writeElement($childElementName, $item);
+    }
+
+}
+
+/**
+ * This function is the 'default' serializer that is able to serialize most
+ * things, and delegates to other serializers if needed.
+ *
+ * The standardSerializer supports a wide-array of values.
+ *
+ * $value may be a string or integer, it will just write out the string as text.
+ * $value may be an instance of XmlSerializable or Element, in which case it
+ *    calls it's xmlSerialize() method.
+ * $value may be a PHP callback/function/closure, in case we call the callback
+ *    and give it the Writer as an argument.
+ * $value may be a an object, and if it's in the classMap we automatically call
+ *    the correct serializer for it.
+ * $value may be null, in which case we do nothing.
+ *
+ * If $value is an array, the array must look like this:
+ *
+ * [
+ *    [
+ *       'name' => '{namespaceUri}element-name',
+ *       'value' => '...',
+ *       'attributes' => [ 'attName' => 'attValue' ]
+ *    ]
+ *    [,
+ *       'name' => '{namespaceUri}element-name2',
+ *       'value' => '...',
+ *    ]
+ * ]
+ *
+ * This would result in xml like:
+ *
+ * <element-name xmlns="namespaceUri" attName="attValue">
+ *   ...
+ * </element-name>
+ * <element-name2>
+ *   ...
+ * </element-name2>
+ *
+ * The value property may be any value standardSerializer supports, so you can
+ * nest data-structures this way. Both value and attributes are optional.
+ *
+ * Alternatively, you can also specify the array using this syntax:
+ *
+ * [
+ *    [
+ *       '{namespaceUri}element-name' => '...',
+ *       '{namespaceUri}element-name2' => '...',
+ *    ]
+ * ]
+ *
+ * This is excellent for simple key->value structures, and here you can also
+ * specify anything for the value.
+ *
+ * You can even mix the two array syntaxes.
+ *
+ * @param Writer $writer
+ * @param string|int|float|bool|array|object
+ * @return void
+ */
+function standardSerializer(Writer $writer, $value) {
+
+    if (is_scalar($value)) {
+
+        // String, integer, float, boolean
+        $writer->text($value);
+
+    } elseif ($value instanceof XmlSerializable) {
+
+        // XmlSerializable classes or Element classes.
+        $value->xmlSerialize($writer);
+
+    } elseif (is_object($value) && isset($writer->classMap[get_class($value)])) {
+
+        // It's an object which class appears in the classmap.
+        $writer->classMap[get_class($value)]($writer, $value);
+
+    } elseif (is_callable($value)) {
+
+        // A callback
+        $value($writer);
+
+    } elseif (is_null($value)) {
+
+        // nothing!
+
+    } elseif (is_array($value)) {
+
+        reset($value);
```

Comment:
Huh weird.. I don't know why `reset` is in there. Perhaps some leftover from an earlier iteration of this. Removing it.